lib/common/transforms.js

```diff
@@ -71,13 +71,16 @@ module.exports = {
   'attribute/cti': {
     type: 'attribute',
     transformer: function(prop) {
-      return {
-        category: prop.path[0],
-        type:     prop.path[1],
-        item:     prop.path[2],
-        subitem:  prop.path[3],
-        state:    prop.path[4]
-      }
+
+      const attrNames = ['category', 'type', 'item', 'subitem', 'state'];
+      const originalAttrs = prop.attributes || {};
+      const generatedAttrs =  {}
+
+      prop.path.forEach((leaf, i) => {
+        generatedAttrs[attrNames[i]] = leaf;
+      });
```

Comment:
if prop.path.length > attrNames.length, then line 80 will set generatedAttrs[undefined] to the last leaf value in prop.path.

This would be new behavior. Maybe the loop should be forced to only run a maximum of 5 times?

Reply:
Great catch. I updated the loop to terminate if either array runs out of values so that you can change the length of the attrNames array in the future. I'll commit that now.

Reply:
Looks great!